scala/scala-impl/src/org/jetbrains/plugins/hydra/compiler/ScalaHydraCompilerConfigurationPanel.scala

```diff
@@ -22,25 +23,58 @@ class ScalaHydraCompilerConfigurationPanel(project: Project, settings: HydraComp
   private val fileChooserDescriptor = new FileChooserDescriptor(false, true, false, false, false, false)
 
   private val documentAdapter = new DocumentAdapter {
-    override def textChanged(documentEvent: DocumentEvent): Unit = downloadButton.setEnabled(getUsername.nonEmpty && getPassword.nonEmpty)
+    override def textChanged(documentEvent: DocumentEvent): Unit =
+      downloadButton.setEnabled(getUsername.nonEmpty && getPassword.nonEmpty && getHydraRepository.nonEmpty && getHydraRepositoryRealm.nonEmpty)
   }
 
   private val focusListener = new FocusListener {
     override def focusGained(e: FocusEvent): Unit = {}
 
     override def focusLost(e: FocusEvent): Unit = if (getUsername.nonEmpty && getPassword.nonEmpty &&
-      (HydraCredentialsManager.getLogin != getUsername || HydraCredentialsManager.getPlainPassword != getPassword)) {
+      (HydraCredentialsManager.getLogin != getUsername ||
+        HydraCredentialsManager.getPlainPassword != getPassword ||
+        hydraGlobalSettings.hydraRepositoryRealm != getHydraRepositoryRealm)) {
+
       HydraCredentialsManager.setCredentials(getUsername, getPassword)
       hydraVersionComboBox.setItems(HydraVersions.downloadHydraVersions)
+      hydraVersionComboBox.setSelectedItem(settings.hydraVersion)
+    }
+  }
+
+  private val urlFocusListener = new FocusListener {
+    override def focusGained(e: FocusEvent): Unit = {}
+
+    override def focusLost(e: FocusEvent): Unit = if (hydraGlobalSettings.getHydraRepositoryUrl.toString != getHydraRepository) {
+      Try(new URL(getHydraRepository)) match {
+        case Success(_) => {
+          hydraGlobalSettings.setHydraRepositopryUrl(getHydraRepository)
+          setHydraRepository(hydraGlobalSettings.getHydraRepositoryUrl.toString)
+          hydraVersionComboBox.setItems(HydraVersions.downloadHydraVersions)
+          hydraVersionComboBox.setSelectedItem(settings.hydraVersion)
+        }
+        case Failure(_) => Messages.showErrorDialog(contentPanel, s"$getHydraRepository is not valid. Default URL will be used.", "URL not valid")
```

Comment:
I don't think the default URL should be used. If the user is editing this field it means he wants to change it, so it makes no sense to use our own default.

Reply:
Yes. But it will not change the field. The field remains with his value and he can change it. Only if he clicks apply then the text field will be changed to the default one. The reason I put that message is to let him know that if he presses the download button the default repo would be used.

Reply:
I'm testing this out and I don't like it. The dialog is very intrusive, I'd prefer the error is displayed when hitting the download button.